include/sparrow/buffer/dynamic_bitset/dynamic_bitset_base.hpp

```diff
@@ -219,7 +226,14 @@ namespace sparrow
         requires std::ranges::random_access_range<std::remove_pointer_t<B>>
     constexpr void dynamic_bitset_base<B>::set(size_type pos, value_type value)
     {
-        SPARROW_ASSERT_TRUE(pos < size());
+         if(pos >= size())
+        {
+            throw std::out_of_range(
+                "dynamic_bitset_base::operator[]: index out of range for dynamic_bitset_base of size"
+                + std::to_string(size()) + " at index " + std::to_string(pos)
+            );
+        }
+        // SPARROW_ASSERT_TRUE(pos < size());
```

Comment:
Same remark here, the assert should be preferred.

Reply:
done